src/main/kotlin/net/lumalyte/lg/interaction/menus/guild/RankEditMenu.kt

```
@@ -58,6 +58,64 @@ class RankEditMenu(private val menuNavigator: MenuNavigator, private val player:
+
+    private fun siblingAt(direction: net.lumalyte.lg.application.services.PriorityDirection): net.lumalyte.lg.domain.entities.Rank? {
+        val siblings = rankService.listRanks(guild.id).sortedBy { it.priority }
+        val idx = siblings.indexOfFirst { it.id == rank.id }
+        val neighborIdx = when (direction) {
+            net.lumalyte.lg.application.services.PriorityDirection.UP -> idx - 1
+            net.lumalyte.lg.application.services.PriorityDirection.DOWN -> idx + 1
+        }
+        return siblings.getOrNull(neighborIdx)
+    }
+
+    private fun addPriorityButtons(pane: StaticPane) {
```

detekt (1.23.8)

[warning] The function addPriorityButtons appears to be too complex based on Cognitive Complexity (complexity: 30). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)